### PR TITLE
Support multiple scaling policies per scalable dimension

### PR DESCRIPTION
**Pull request by surajkota**
## Description of changes:
Customer facing issue in using multiple scaling policies for a scalable dimension because originally the code assumed it is 1:1 mapping. To uniquely identify a scaling policy resource, we need policy name, resource-id, service namespace and scalable dimension. This PR adds ability to create multiple scaling policies per scalable dimension and also updates the following things:

- Pass PolicyName for sdkFind, it is a required field in CRD
- Added logic to error on missing required fields per description above for DescribeScalingPolicyInput
- Updated tests to cover multiple scaling policies scenario per dimension case
- Updated test logic to use correct dependency of fixtures and order of execution 
- Updates the tests to use object oriented style bootstrapping. Hit https://github.com/aws-controllers-k8s/test-infra/pull/368 issue while trying to fix the issue and cannot use latest ack-test without updating the bootstrap logic
- Removed duplicate test with dynamodb. SageMaker and dynamodb tests are testing same thing while SM tests have more coverage. No reason to maintain both set of tests, dynamodb tests add additional bootstrapping as well

~~**:warning: Pending before merge**~~
Hit another issue described in https://github.com/aws-controllers-k8s/community/issues/1871. Fix https://github.com/aws-controllers-k8s/code-generator/pull/459 has been merged

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/service_bootstrap.py [hunk 1]
```diff
@@ -22,102 +22,44 @@
...
+    account_id = get_account_id()
+    bucket_name = f"ack-test-bucket-{region}-{account_id}"
+
+    resources = TestBootstrapResources(
+        DataBucket=Bucket(bucket_name),
+        SageMakerExecutionRole=Role(
+            "ack-sagemaker-execution-role",
+            "sagemaker.amazonaws.com",
+            managed_policies=[
+                "arn:aws:iam::aws:policy/AmazonSageMakerFullAccess",
```
Comment (ryansteakley) [inline test/e2e/service_bootstrap.py:47]: Is sagemaker and s3 fullaccess sufficient for autoscaling? Looks like that was what was there previously, but double-checking.

File: helm/values.yaml [hunk 2]
```diff
@@ -90,7 +110,7 @@ deletionPolicy: delete
 # controller reconciliation configurations
 reconcile:
   # The default duration, in seconds, to wait before resyncing desired state of custom resources.
-  defaultResyncPeriod: 0
```
Comment (ryansteakley) [inline helm/values.yaml:93]: Reasoning for this ?

File: helm/templates/deployment.yaml
Comment (ryansteakley) [inline helm/templates/deployment.yaml:1]: Are these changes part of new code-gen or added personally?

**Review by ryansteakley — APPROVED** [review]

File: helm/templates/deployment.yaml
Comment (surajkota) [reply helm/templates/deployment.yaml:1]: changed from latest code generator

File: helm/values.yaml (hunk 2 above)
Comment (surajkota) [reply helm/values.yaml:93]: changed from latest code generator

**Review by RedbackThomson — APPROVED** [review]
A lot to follow, given there are code gen changes, updates to the testing framework and pending changes. I would have preferred we split these out, so that we can review each separately (and make it easier to revert if need be in the future).

Changes do seem to be pretty self explanatory, though

File: test/e2e/service_bootstrap.py (hunk 1 above)
Comment (surajkota) [reply test/e2e/service_bootstrap.py:47]: yes. SageMakerFullAccess has all the permissions reqd for autoscaling

**Pull request merged**